packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts

```diff
@@ -371,47 +371,50 @@ describe('plugin-metrics', () => {
 
       describe('when the request fails', () => {
         it('does not clear the queue', async () => {
+          // sinon appears to have gap in its api where stub.onCall(n) doesn't
+          // accept a function, so the following is more verbose than one might
+          // desire
+          webex.request = function () {
+            // noop
+          };
+
+          sinon.stub(webex, 'request').callsFake(() => {
+            return Promise.reject("error")
+          });
+
           // avoid setting .sent timestamp
           webex.internal.newMetrics.callDiagnosticMetrics.callDiagnosticEventsBatcher.prepareRequest =
             (q) => Promise.resolve(q);
 
-          const err = new Error('error');
-          webex.request = sinon.stub().returns(Promise.reject(err));
-
           webex.logger.error = sinon.stub();
           webex.logger.log = sinon.stub();
           sinon.stub(Utils, 'generateCommonErrorMetadata').returns('formattedError');
 
+
           const promise = webex.internal.newMetrics.callDiagnosticMetrics.submitToCallDiagnostics({
             event: 'my.event',
           });
 
-          await flushPromises();
-          clock.tick(config.metrics.batcherWait);
-
-          let error;
-
-          // catch the expected error and store it
-          await promise.catch((e) => {
-            error = e;
-          });
-
-          // This is horrific, but stubbing lodash is proving difficult
-          const expectedBatchId = parseInt(uniqueId()) - 1;
-
-          // check that promise was rejected with the original error of the webex.request
-          assert.deepEqual(err, error);
-
-          assert.calledOnceWithExactly(
-            webex.logger.error,
-            'call-diagnostic-events -> ',
-            `CallDiagnosticEventsBatcher: @submitHttpRequest#ca-batch-${expectedBatchId}. Request failed:`,
-            `error: formattedError`
-          );
-          assert.lengthOf(
-            webex.internal.newMetrics.callDiagnosticMetrics.callDiagnosticEventsBatcher.queue,
-            0
-          );
+          return promiseTick(100)
```

Comment:
note: when I mentioned to you about the draft PR where I fixed the test, I wasn't suggesting to revert to use promiseTick approach, I was only referring to asserting the log using sinon.match (to avoid the lodash mock issue)